src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1478,6 +1478,39 @@ public HashSet<(uint256, DateTimeOffset)> RemoveAllTransactions(string walletNam
             return removedTransactions;
         }
 
+        /// <inheritdoc />
+        public HashSet<(uint256, DateTimeOffset)> RemoveTransactionsFromDate(string walletName, DateTimeOffset fromDate)
+        {
+            Guard.NotEmpty(walletName, nameof(walletName));
+            Wallet wallet = this.GetWallet(walletName);
+
+            var removedTransactions = new HashSet<(uint256, DateTimeOffset)>();
+
+            lock (this.lockObject)
+            {
+                IEnumerable<HdAccount> accounts = wallet.GetAccountsByCoinType(this.coinType);
+                foreach (HdAccount account in accounts)
+                {
+                    foreach (HdAddress address in account.GetCombinedAddresses())
+                    {
+                        var toRemove = address.Transactions.Where(t => t.CreationTime > fromDate).ToList();
+                        foreach (var trx in toRemove)
+                        {
+                            removedTransactions.Add((trx.Id, trx.CreationTime));
+                            address.Transactions.Remove(trx);
+                        }
+                    }
+                }
+            }
+
+            if (removedTransactions.Any())
+            {
+                this.SaveWallet(wallet);
+            }
+
+            return removedTransactions;
```

Comment:
What's the point of returning the set of removed transactions @bokobza ?

Reply:
So the UI can display what has been removed, or even display the number of removed transactions.

Reply:
Ah yes coo!